src/bidiMapper/utils/cdpServer.spec.ts

```diff
@@ -0,0 +1,107 @@
+ */
+import { connectCdp } from './cdpClient';
+import { StubServer } from './tests/stubServer.spec';
+import { assert as chai_assert } from 'chai';
+import { assert as sinon_assert } from 'sinon';
+import { Protocol } from 'devtools-protocol';
+
+describe('CdpClient tests.', async () => {
+  it(`given CdpClient, when some command is called, then cdpBindings should be
+      called with proper values`, async () => {
+    const expectedMessageStr = JSON.stringify({
+      id: 0,
+      method: 'Target.activateTarget',
+      params: {
+        targetId: 'TargetID',
+      },
+    });
+
+    const mockCdpServer = new StubServer();
+
+    const cdpClient = connectCdp(mockCdpServer);
+    cdpClient.Target.activateTarget({ targetId: 'TargetID' });
+
+    sinon_assert.calledOnceWithExactly(
+      mockCdpServer.sendMessage,
+      expectedMessageStr
+    );
+  });
+
+  it(`given some command is called, when CDP command is done, then
+        'sendMessage' promise is resolved`, async () => {
+    const mockCdpServer = new StubServer();
+    const cdpClient = connectCdp(mockCdpServer);
+
+    // Get handler 'onMessage' to notify 'cdpClient' about new CDP messages.
+    const onMessage = mockCdpServer.getOnMessage();
+
+    // Send CDP command and store result promise.
+    const commandPromise = cdpClient.Target.activateTarget({
+      targetId: 'TargetID',
+    });
+
+    // Verify CDP command was sent.
+    sinon_assert.calledOnce(mockCdpServer.sendMessage);
+
+    // Notify 'cdpClient' the CDP command is finished.
+    onMessage(JSON.stringify({ id: 0, result: {} }));
+
+    // Assert 'cdpClient' resolved message promise.
+    await commandPromise;
+  });
+
+  it(`given some command is called 2 times, when CDP commands are done, then
+        each command promise is resolved with proper results`, async () => {
+    const mockCdpServer = new StubServer();
+    const cdpClient = connectCdp(mockCdpServer);
+    const expectedResult1 = {
+      someResult: 1,
+    } as any as Protocol.Target.AttachToTargetResponse;
+    const expectedResult2 = {
+      anotherResult: 2,
+    } as any as Protocol.Target.AttachToTargetResponse;
+    const commandResult1 = { id: 0, result: expectedResult1 };
+    const commandResult2 = { id: 1, result: expectedResult2 };
+
+    // Get handler 'onMessage' to notify 'cdpClient' about new CDP messages.
+    const onMessage = mockCdpServer.getOnMessage();
+
+    // Send 2 CDP commands and store result promise.
+    const commandPromise1 = cdpClient.Target.attachToTarget({
+      targetId: 'Target_1',
+    });
+    const commandPromise2 = cdpClient.Target.attachToTarget({
+      targetId: 'Target_2',
+    });
+
+    // Verify CDP command was sent.
+    sinon_assert.calledTwice(mockCdpServer.sendMessage);
+
+    // Notify 'cdpClient' the command2 is finished.
+    onMessage(JSON.stringify(commandResult2));
+    // Assert second message promise is resolved.
+    const actualResult2 = await commandPromise2;
```

Comment:
The `await` here does mean that the test is verifying that commands aren't in a blocking queue, which is great.

Could we also verify that the order that the promises resolve is correct? This test could also pass if `commandPromise1` is incorrectly resolved at this point.

Reply:
It's implicit by the fact the `commandPromise1` should be resolved with the `commandResult1`, which, in order, was introduced after the `commandPromise2` resolved. 

If you know an easy way to verify the promise is not resolved, I'd be happy to add it. But al the ways I can think of are quite bulky.